api/src/main/java/org/openmrs/customdatatype/Customizable.java

```diff
@@ -11,39 +11,116 @@
 
 import java.util.Collection;
 import java.util.List;
+import java.util.stream.Collectors;
 
 import org.openmrs.attribute.Attribute;
 
 /**
- * Marker interface for classes that may be customized by the user by adding custom attributes, e.g. Visit
- * has VisitAttributes, so it implements {@link Customizable}&lt;VisitAttribute&gt;
+ * Marker interface for classes that may be customized by the user by adding custom attributes, e.g.
+ * Visit has VisitAttributes, so it implements {@link Customizable}&lt;VisitAttribute&gt;
+ * 
  * @param <A> the type of attribute held
  * @since 1.9
  */
 @SuppressWarnings("rawtypes")
 public interface Customizable<A extends Attribute> {
 	
 	/**
-	 * @return all attributes (including voided ones)
+	 * @return all attributes (including voided ones), must not be null
 	 */
 	Collection<A> getAttributes();
 	
 	/**
+	 * @param attributes the attributes to set
+	 */
+	void setAttributes(Collection<A> attributes);
+	
+	/**
 	 * @return non-voided attributes
+	 * @should return empty collection if attributes collection is empty
+	 * @should return only non voided attributes
 	 */
-	Collection<A> getActiveAttributes();
+	public default Collection<A> getActiveAttributes() {
+		return getAttributes().stream()//
+		        .filter(attr -> !attr.getVoided())//
```

Comment:
@teleivo yes, sure, you can assign it to me. I started taking a look, but I think its a bit of a complicated issue. 
I think it would be best to choose some sample classes, try different formatter settings and let the community vote on them, what do you think?